### ISSUE TITLE
Redux state for the `statusmessage` cookie

### ISSUE THREAD

**Issue by JeffersonBledsoe** [plone/volto#4355]
**Is your feature request related to a problem? Please describe.**
Some Plone responses set a `statusmessage` cookie. This cookie is the 'raw' value as base64 to be used by products.statusmessage which does a [little bit of parsing to split apart the message and status type](https://github.com/plone/Products.statusmessages/blob/bc249389485dec8c7e3ed64c617bd5263fe55de0/Products/statusmessages/message.py). It would be nice to encapsulate the logic that is normally done in classic in Volto too and provide the status messages in an easier to use format for any potential Volto users.

**Describe the solution you'd like**
- A redux state that is updated when the `statusmessage` cookie changes. The reducer performs the same message/ type splitting done by products.statusmessage and presents the message/ type in a format more useable in JS.

**Describe alternatives you've considered**
- Having the decryption function as a helper may also suffice, however this means developers will need to manually wire up the cookie listening/ updating logic themselves

**Additional context**
Whilst plone.restapi doesn't currently use status messages (see plone/plone.restapi#584), some addons still provide useful information via IStatusMessage.

**Comment by tiberiuichim** [comment]
I guess that would require some type of cookie watcher, like https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/onChanged ?

**Comment by JeffersonBledsoe** [comment]
> I guess that would require some type of cookie watcher, like https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/onChanged ?

Yep. We would probably use `useCookies(['statusmessage'])` from [react-cookie](https://www.npmjs.com/package/react-cookie) or something similar to handle this as Safari support isn't there and I'm not sure on the state of polyfills for the API. The `Cookies Store` API looks like a similar situation.
